R/calculate_pga.R

```diff
@@ -130,23 +152,47 @@ calculate_pga = function(this_seg,
 
   # calculate total length of CNV
   affected_regions = this_seg %>%
-    dplyr::mutate(size = end - start) %>%
-    group_by(sample) %>%
-    summarise(total = sum(size))
+    dplyr::mutate(size = end - start)
+
+  if(per_chromosome){
+   affected_regions <- affected_regions %>%
+    group_by(sample, chrom) %>%
+    summarise(total = sum(size),
+              chrom_len = unique(chr_len))
+   affected_regions$PGA.chr = affected_regions$total/affected_regions$chrom_len
+     # add chr with 0
+   affected_regions = affected_regions %>%
+    dplyr::select(!all_of(c("total", "chrom_len"))) %>%
+    dplyr::mutate(PGA.chr = round(PGA.chr, 3)) %>%
+    pivot_wider(names_from = chrom, values_from = PGA.chr)
+
+      # now add any samples that can have 0 PGA
+   affected_regions = base::merge(sample_set,
+                                 affected_regions,
+                                 all.x = TRUE)
 
-  affected_regions$PGA = affected_regions$total / genome_size
+   affected_regions[is.na(affected_regions)] <- 0
 
-  # now add any samples that can have 0 PGA
-  affected_regions = base::merge(sample_set,
+   affected_regions = affected_regions %>%
+    rename("sample_id" = "sample")
+   colnames(affected_regions)[2:ncol(affected_regions)] <- paste0("chr",colnames(affected_regions)[2:ncol(affected_regions)],"_pga")
+  } else {
+   affected_regions <- affected_regions %>%
+        group_by(sample) %>%
```

Comment:
The following warning:
```
`summarise()` has grouped output by 'sample'. You can override using the `.groups` argument.
```
Can be avoided by first doing the `summarise` then `group_by`.

Reply:
Are you getting this in the above example you posted above? I am not seeing this warning in neither version (per chromosome or not), and as well as when trying the version on master. What is the dplyr version you have? Mine is 1.1.1 and it is probably the culprit. I can add this change but just want to make sure it won't have unexpected behaviour, and since I don;t have this warning can't confirm the change behaves as expected

Reply:
My worry is that if I do first summarise followed by group_by like you suggested it will collapse the data for all patients in the input, and every single sample will have the exact same value, representing the mean across all samples